tests/rules/test_check_pr_target.py

```diff
@@ -0,0 +1,277 @@
+    workflow = """\
+---
+on:
+  workflow_dispatch:
+  pull_request_target:
+    types: [opened, synchronize]
+    branches:
+      - 'main'
+      - 'not main'
+
+jobs:
+  check-run:
+    name: Check PR run
+    uses: bitwarden/gh-actions/.github/workflows/check-run.yml@main
+  
+  quality:
+    name: Quality scan
+    needs: check-run
+    steps:
+      - run: echo test
+ 
+  dependent-job:
+     name: Another Dependent Job
+     needs:
+       - quality
+       - check-run
+     steps:
+       - run: echo another dependent job
+"""
+    return WorkflowBuilder.build(workflow=yaml.load(workflow), from_file=False)
+
+@pytest.fixture(name="two_failures_workflow")
+def fixture_two_failures_workflow():
+    workflow = """\
+---
+on:
+  workflow_dispatch:
+  pull_request_target:
+    types: [opened, synchronize]
+
+jobs:
+  check-run:
+    name: Check PR run
+    uses: bitwarden/some-other-repo/.github/workflows/check-run.yml@main
+  
+  quality:
+    name: Quality scan
+    needs: check-run
+    steps:
+      - run: echo test
+ 
+  dependent-job:
+     name: Another Dependent Job
+     needs:
+       - quality
+       - check-run
+     steps:
+       - run: echo another dependent job
+"""
+    return WorkflowBuilder.build(workflow=yaml.load(workflow), from_file=False)
+
+@pytest.fixture(name="rule")
+def fixture_rule():
+    return RuleCheckPrTarget()
+
+def test_rule_on_correct_workflow(rule, correct_workflow):
+    result, message = rule.fn(correct_workflow)
+    assert result is True
+    assert message == ""
+
+
+def test_rule_on_no_checkworkflow(rule, no_check_workflow):
+    result, message = rule.fn(no_check_workflow)
+    assert result is False
+    assert message == message
+
+def test_rule_on_no_target_workflow(rule, no_target_workflow):
+    result, message = rule.fn(no_target_workflow)
+    assert result is True
+    assert message == ""
+
+def test_rule_on_jobs_without_needs(rule, no_needs_workflow):
+    result, message = rule.fn(no_needs_workflow)
+    assert result is False
+    assert message == message, "check-run is missing from the following jobs in the workflow: quality"
+
+def test_rule_on_dependencies_without_check(rule, dependent_missing_check_workflow):
+    result, message = rule.fn(dependent_missing_check_workflow)
+    assert result is False
+    assert message == message, "check-run is missing from the following jobs in the workflow: dependent-job"
+
+def test_rule_on_no_target_workflow(rule, no_branches_workflow):
+    result, message = rule.fn(no_branches_workflow)
+    assert result is False
+    assert message == "Workflows using pull_request_target can only target the main branch"
+
+def test_rule_on_no_target_workflow(rule, bad_branches_workflow):
+    result, message = rule.fn(bad_branches_workflow)
+    assert result is False
+    assert message == "Workflows using pull_request_target can only target the main branch"
+
+def test_rule_on_no_target_workflow(rule, two_failures_workflow):
```

Comment:
duplicate name just needs changed
```suggestion
def test_rule_on_two_failures(rule, two_failures_workflow):
```